src/jaegertracing/baggage/BaggageTest.cpp

```diff
@@ -105,7 +157,19 @@ TEST(Baggage, testRemoteRestrictionManagerFunctionality)
     ASSERT_EQ(Restriction(false, 0),
               manager.getRestriction("test-service", "abc"));
     mockAgent->addBaggageRestriction("abc", Restriction(true, 1));
-    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+    // Wait for the Update Restrictions to occur
+    // This is done by listening to the number of update published via the Metrics
+    // Typically, On Windows with Debug mode, the update happens after 700ms
+    for (int sleepIndex = 0; sleepIndex < 10; ++sleepIndex) {
+        auto counterVal = testReporter.counterValue(
+            "jaeger.baggage-restrictions-update", { { "result", "ok" } });
+        std::this_thread::sleep_for(std::chrono::milliseconds(200));
```

Comment:
Probably don't need to sleep if `counterValue` waits on mutex.

Reply:
It's just a mutex and not a condition variable. We can have a spinning loop but we'll have to handle the timeout.

Reply:
OK I understand the idea. If you'd like, you can add a condition variable too. Otherwise, this makes sense. Would prefer the condition variable to avoid sleeps, but will accept either solution.